src/EventSubscriber/Api/Oauth2UserResolveSubscriber.php

```diff
@@ -24,6 +28,11 @@ public function checkUserPasswordValidity(UserResolveEvent $event): UserResolveE
         $isUserPasswordInvalid = $user instanceof LegacyPasswordAuthenticatedUserInterface && !$this->hasher->isPasswordValid($user, $event->getPassword());
         $event->setUser($isUserPasswordInvalid ? null : $user);
 
+        if (!$user->hasPasswordWithLatestPolicy() && $this->passwordHelper->isStrongPassword($event->getPassword())) {
+            $user->setHasPasswordWithLatestPolicy(true);
+            $this->em->flush();
+        }
+
         return $event;
```

Comment:
I think we also have this code in the authenticator class. Maybe we could use a service or something ?

Reply:
Like `AuthenticationProcess` and centralize all checks we do on the user during authentication on web or api